torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp

```diff
@@ -207,9 +110,19 @@ std::shared_ptr<PointwiseParams> getPointwiseHeuristics(
 
   auto in_tvs = ir_utils::filterByType<TensorView>(fusion->inputs());
 
-  DomainMap domain_map(fusion);
-
-  TensorView* largest_out = domain_map.findReferenceTensorView();
+  auto domain_map_entry =
+      HeuristicSummaryEntry<HeuristicCompileTime::DomainMap>(
+          data_cache,
+          [fusion]() { return std::make_unique<DomainMap>(fusion); });
+  const auto& domain_map = dynamic_cast<DomainMap&>(domain_map_entry.get());
+
+  auto largest_out_entry =
+      HeuristicSummaryEntry<HeuristicCompileTime::ReferenceTensors>(
+          data_cache, [&domain_map]() {
+            std::vector<TensorView*> data{domain_map.findReferenceTensorView()};
+            return std::make_unique<std::vector<TensorView*>>(std::move(data));
+          });
+  TensorView* largest_out = largest_out_entry.get()[0];
 
```

Comment:
Thanks for caching these entries here. This path looks lightweight enough to me now.